ovis_demo/src/ovis_demo.cpp

```diff
@@ -0,0 +1,318 @@
+#include <algorithm>
+#include <string>
+#include <iostream>
+#include <vector>
+
+#include "ovis_demo.hpp"
+
+#include <ros/ros.h>
+#include <sensor_msgs/Joy.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/client/simple_client_goal_state.h>
+#include <kinova_msgs/ArmJointAnglesAction.h>
+#include <Kinova.API.UsbCommandLayerUbuntu.h>
+
+#include <inttypes.h>
+#include <string>
+#include <stdexcept>
+#include <dlfcn.h>
+
+constexpr uint8_t NUMBER_OF_JOINTS = 6;
+constexpr uint8_t NUMBER_OF_DEGREE_PER_GOAL = 1;
+
+static uint8_t joint_index;
+static uint8_t joint_active[NUMBER_OF_JOINTS];
+static bool bumper_left_pressed = false;
+static bool bumper_right_pressed = false;
+static AngularPosition joints_angles;
+static TrajectoryPoint trajectory_point;
+
+static unsigned int const LEFT_BUMPER_INDEX = 4;
+static unsigned int const RIGHT_BUMPER_INDEX = 5;
+
+void sendJointCommand(std::vector<float> const joy_axes)
+{
+  // Act as a deadzone for the gamepad
```

Comment:
TODO : Remove deadzone, this is defined in the controller launch file. See : http://wiki.ros.org/joy?distro=melodic